frame/bounties/src/lib.rs

```diff
@@ -660,12 +785,685 @@ decl_module! {
+						let fee = subbounty.fee.min(balance); // just to be safe
+						let payout = balance.saturating_sub(fee);
+
+						// unreserve the subcurator deposit
+						let _ = T::Currency::unreserve(
+							&subcurator,
+							subbounty.curator_deposit,
+						); // should not fail
+
+						// Make payout to subcurator
+						let _ = T::Currency::transfer(
+							&subbounty_account,
+							&subcurator,
+							fee,
+							AllowDeath,
+						); // should not fail
+
+						// Make payout to beneficiary
+						let _ = T::Currency::transfer(
+							&subbounty_account,
+							beneficiary,
+							payout,
+							AllowDeath,
+						); // should not fail
+
+						// Trigger the SubBountyClaimed event
+						Self::deposit_event(
+							Event::<T>::SubBountyClaimed(
+								bounty_id,
+								subbounty_id,
+								payout,
+								beneficiary.clone(),
+							)
+						);
+
+						// Update the active subbounty tracking count.
+						Bounties::<T>::mutate_exists(
+							bounty_id,
+							|maybe_bounty| {
+								if let Some(bounty) = maybe_bounty.as_mut() {
+									bounty.active_subbounty_count = bounty
+										.active_subbounty_count
+										.saturating_sub(1);
+								}
+							}
+						);
+						// Remove the subbounty description
+						BountyDescriptions::remove(subbounty_id);
+						// Remove the subbounty instance
+						// from DB
+						*maybe_subbounty = None;
+						Ok(())
+					} else {
+						Err(Error::<T>::UnexpectedStatus.into())
+					}
+				}
+			)?;
+		}
+
+		/// Cancel a proposed or active subbounty.
+		/// Subbounty account funds are transferred to parent bounty account.
+		/// The subcurator deposit may be unreserved if possible.
+		///
+		/// The dispatch origin for this call must be
+		/// either master curator of this subbounty or `T::RejectOrigin`.
+		///
+		/// If the state of subbounty is `Active`,
+		/// subcurator deposit is unreserved.
+		///
+		/// If the state of subbounty is `PendingPayout`,
+		/// call fails & returns `PendingPayout` error.
+		///
+		/// For the origin other than T::RejectOrigin,
+		/// Parent bounty must be in active state,
+		/// for this subbounty call to work. For origin
+		/// T::RejectOrigin execution is forced.
+		///
+		/// Instance of subbounty is deallocated from DB
+		/// on successful call completion.
+		///
+		/// - `bounty_id`: ID pair Bounty ID.
+		/// - `subbounty_id`: ID pair SubBounty ID to cancel.
+		#[weight = <T as Config>::WeightInfo::close_subbounty()]
+		fn close_subbounty(origin,
+			#[compact] bounty_id: BountyIndex,
+			#[compact] subbounty_id: BountyIndex,
+		) {
+			let maybe_sender = ensure_signed(origin.clone())
+				.map(Some)
+				.or_else(|_| T::RejectOrigin::ensure_origin(origin).map(|_| None))?;
+
+			// Ensure parent bounty exist, get master_curator or default accountid.
+			let maybe_bounty_active_status = Self::ensure_bounty_exist(
+				bounty_id,
+				maybe_sender.is_none(),
+			)?;
+
+			ensure!(
```

Comment:
yeah this can def. be a helper function

Reply:
Sorry didn't get this part Kian

do you want to keep, below expression as separate helper function ?

```
			ensure!(
				maybe_sender.map_or(
					true,
					|sender| maybe_bounty_active_status.map_or(true, |v| v.0 == sender),
				),
				BadOrigin,
			);

```

Reply:
yeah maybe, similar to what I said in https://github.com/paritytech/substrate/pull/7965#discussion_r626425056 because it is repeated.